modules/catalog/src/main/scala/lucuma/catalog/csv/TargetImport.scala

```diff
@@ -253,22 +243,73 @@ object TargetImport:
+        .collectFirst(row.as[A](_))
+        .getOrElse(DecoderError(s"No column $col found").asLeft)
+
+    private def internalAlternatives[T](
+      alternatives: List[String],
+      defaultValue: Option[T] = None
+    )(using CellDecoder[Option[T]]): DecoderResult[Option[T]] = {
+      val (errors, correct) = alternatives.toList
+        .map(row.as[Option[T]](_))
+        .partitionEither(identity)
+      if (correct.nonEmpty) correct.head.asRight
+      else if (errors.nonEmpty) {
+        val notFound = errors.forall { case d: DecoderError =>
+          d.getMessage.startsWith("unknown field")
+        }
+        // if not found anywhere go to the default
+        if (notFound)
+          defaultValue.asRight
+        else
+          errors.head.withLine(row.line).asLeft
+      } else DecoderError(s"No column col found").withLine(row.line).asLeft
+    }
+
+    // Try to decode a column with several alternative capitalizations
+    // If any fails the whole thing fails, if none is found return none
+    def withAlternatives[T](
+      col:          String,
+      defaultValue: Option[T] = None
+    )(using CellDecoder[Option[T]]): DecoderResult[Option[T]] =
+      internalAlternatives(List(col, col.toUpperCase(), col.toLowerCase(), col.capitalize))
+
+    //
+    // Try to decode a column with several alternative each expanded to several capitalization
+    // If any fails the whole thing fails, if none is found return none
+    def withAlternativesM[T](
+      defaultValue: Option[T] = None,
+      col:          String,
+      extras:       String*
+    )(using CellDecoder[Option[T]]): DecoderResult[Option[T]] =
+      internalAlternatives(
+        (col :: extras.toList).flatMap(col =>
+          List(col, col.toUpperCase(), col.toLowerCase(), col.capitalize)
+        )
+      )
+
   given CsvRowDecoder[TargetCsvRow, String] =
     (row: CsvRow[String]) =>
       for {
-        name              <- row.as[NonEmptyString]("Name").orElse(row.as[NonEmptyString]("NAME"))
-        ra                <- row.as[Option[RightAscension]]("RAJ2000").defaultToNone(row)
-        dec               <- row
-                               .as[Option[Declination]]("DecJ2000")
-                               .orElse(row.as[Option[Declination]]("DECJ2000"))
-                               .defaultToNone(row)
-        pmRa              <- row
-                               .as[Option[ProperMotion.RA]]("pmRa")
-                               .defaultToNone(row)
-        pmDec             <- row
-                               .as[Option[ProperMotion.Dec]]("pmDec")
-                               .defaultToNone(row)
-        epoch             <- row.as[Option[Epoch]]("epoch").defaultToNone(row)
+        name              <- row.asIn[NonEmptyString]("Name", "NAME")
+        ra                <- row.withAlternativesM[RightAscension](None, "RAJ2000", "RaJ2000")
+        dec               <- row.withAlternativesM[Declination](None, "DecJ2000", "DECJ2000")
```

Comment:
Wouldn't using "RaJ2000" and "DecJ2000" with the regular `withAlternatives` (not the `M` version) catch the other alternatives? Or, maybe `raJ2000` and `decJ2000` would catch more, since `capitalize` would create the former.

Reply:
well that's right for `DecJ200` not so for `Ra` and `RA`